### PR TITLE
Add reflective boundary conditions for `BBMBBMEquations1D`

### PR DESCRIPTION
**Pull request by JoshuaLampert**
As a start for supporting reflecting bc, I implemented them for the `BBMBBMEquations1D` as proposed in [H. Ranocha, D. Mitsotakis, D. Ketcheson, A Broad Class of Conservative Numerical Methods for Dispersive Wave Equations (2021)](https://arxiv.org/abs/2006.14802).
Convergence study looks fine, I think. We get something like $p/2 + 1$ for $\eta$ and $p$ for $v$, which is expected due to the reduced order at the boundary, right @ranocha?
<details>
  <summary>p = 2:</summary>

  ```julia
julia> convergence_test("examples/bbm_bbm_1d/bbm_bbm_1d_basic_reflecting.jl", 4, N = 16, accuracy_order = 2)
[...]
####################################################################################################
l2
η                        v                        
N    error     EOC       N    error     EOC       
16   1.18e-01  -         16   1.01e-01  -         
32   2.68e-02  2.14      32   2.26e-02  2.15      
64   6.43e-03  2.06      64   5.43e-03  2.06      
128  1.58e-03  2.03      128  1.33e-03  2.03      

mean           2.08      mean           2.08      
----------------------------------------------------------------------------------------------------
linf
η                        v                        
N    error     EOC       N    error     EOC       
16   2.37e-01  -         16   2.15e-01  -         
32   5.15e-02  2.20      32   4.87e-02  2.14      
64   1.23e-02  2.07      64   1.18e-02  2.05      
128  3.00e-03  2.03      128  2.89e-03  2.03      

mean           2.10      mean           2.07      
----------------------------------------------------------------------------------------------------
  ```
</details>

<details>
  <summary>p = 4:</summary>

```julia
julia> convergence_test("examples/bbm_bbm_1d/bbm_bbm_1d_basic_reflecting.jl", 4, N = 16, accuracy_order = 4)
[...]
####################################################################################################
l2
η                        v                        
N    error     EOC       N    error     EOC       
16   2.15e-02  -         16   3.16e-03  -         
32   4.26e-03  2.34      32   2.07e-04  3.93      
64   6.49e-04  2.71      64   1.16e-05  4.15      
128  8.78e-05  2.88      128  6.74e-07  4.11      

mean           2.65      mean           4.07      
----------------------------------------------------------------------------------------------------
linf
η                        v                        
N    error     EOC       N    error     EOC       
16   5.38e-02  -         16   7.70e-03  -         
32   9.30e-03  2.53      32   4.83e-04  4.00      
64   1.30e-03  2.84      64   2.62e-05  4.20      
128  1.69e-04  2.94      128  1.47e-06  4.15      

mean           2.77      mean           4.12      
----------------------------------------------------------------------------------------------------
```
</details>

<details>
  <summary>p = 6:</summary>

```julia
julia> convergence_test("examples/bbm_bbm_1d/bbm_bbm_1d_basic_reflecting.jl", 4, N = 16, accuracy_order = 6)
[...]
####################################################################################################
l2
η                        v                        
N    error     EOC       N    error     EOC       
16   1.76e-01  -         16   2.87e-03  -         
32   1.35e-02  3.71      32   3.60e-05  6.32      
64   7.96e-04  4.08      64   6.38e-07  5.82      
128  4.67e-05  4.09      128  2.73e-08  4.55      

mean           3.96      mean           5.56      
----------------------------------------------------------------------------------------------------
linf
η                        v                        
N    error     EOC       N    error     EOC       
16   3.63e-01  -         16   7.56e-03  -         
32   3.17e-02  3.52      32   1.13e-04  6.06      
64   2.00e-03  3.98      64   1.69e-06  6.07      
128  1.22e-04  4.04      128  5.02e-08  5.07      

mean           3.85      mean           5.73      
----------------------------------------------------------------------------------------------------
```
</details>

### CODE REVIEW

File: examples/bbm_bbm_1d/bbm_bbm_1d_basic_reflecting.jl [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+# create solver with SBP operators of accuracy order 4
+accuracy_order = 4
+D1 = derivative_operator(MattssonNordström2004(),
+                         derivative_order = 1, accuracy_order = accuracy_order,
+                         xmin = mesh.xmin, xmax = mesh.xmax, N = mesh.N)
+D2 = derivative_operator(MattssonNordström2004(),
+                         derivative_order = 2, accuracy_order = accuracy_order,
+                         xmin = mesh.xmin, xmax = mesh.xmax, N = mesh.N)
+solver = Solver(D1, D2)
```
Comment (ranocha) [inline examples/bbm_bbm_1d/bbm_bbm_1d_basic_reflecting.jl:28]: `D2` is not used here, is it? Can we clarify this by passing `nothing` or something like that instead - or add a comment?

**Review by ranocha — CHANGES_REQUESTED** [review]
Thanks a lot! This looks quite good to me. And yes, the EOC is reasonable 👍

File: examples/bbm_bbm_1d/bbm_bbm_1d_basic_reflecting.jl (hunk 1 above)
Comment (JoshuaLampert) [reply examples/bbm_bbm_1d/bbm_bbm_1d_basic_reflecting.jl:28]: `D2` is used [for the Dirichlet boundary condition](https://github.com/JoshuaLampert/DispersiveShallowWater.jl/pull/104/files#diff-014e2648af7ba655effcd1aaea79591f8e82335301f5be67fecfc6c80ecb834bR150).

Comment (ranocha) [reply examples/bbm_bbm_1d/bbm_bbm_1d_basic_reflecting.jl:28]: Right, I just thought about the Neumann one. Sorry

**Review by ranocha — APPROVED** [review]

**Pull request merged**